packages/web-app-files/src/mixins/actions/restore.js

```diff
@@ -58,7 +58,14 @@ export default {
         restorePromises.push(
           restoreQueue.add(async () => {
             try {
-              await this.$client.fileTrash.restore(path, resource.id, restorePath)
+              const len = path.split('/').length
+              const foldersToRestore = restorePath.split('/')
+              foldersToRestore.splice(0, len)
+              foldersToRestore.pop()
+              for (const folder of foldersToRestore) {
+                await this.$client.files.createFolder(buildWebDavFilesPath(this.user.id, folder))
+              }
```

Comment:
This will not work with deep nested folders yet. Let's say you have the following folder structure: `level1/level2/level3`. `foldersToRestore` would be an array including all folders: `['level1', 'level2', 'level3']`. By just looping through them you will create those in a flat hierarchy.

You will have to add the created folders to the path you pass to `createFolder()`. E.g. first call `createFolder('level1')`, next `createFolder('level1/level2')` etc.

Reply:
ah yes you are right, but want to make sure the edge case commented in the issue is cleared, do you maybe have an idea/understand what the problem there is :)

Reply:
I've commented in https://github.com/owncloud/web/issues/1753#issuecomment-1123762599. Quiet a nasty edge case :(